runtime/src/main/kotlin/kotlin/text/regex/Lexer.kt

```diff
@@ -457,7 +457,7 @@ internal class Lexer(val patternString: String, flags: Int) {
             // Word/whitespace/digit.
             'w', 's', 'd', 'W', 'S', 'D' -> {
                 lookAheadSpecialToken = AbstractCharClass.getPredefinedClass(
-                        String(pattern, prevNonWhitespaceIndex, 1),
+                        pattern.concatToString(prevNonWhitespaceIndex, 1),
```

Comment:
It looks like it should be `pattern.concatToString(prevNonWhitespaceIndex, prevNonWhitespaceIndex+1)`